gp-templates/helper-functions.php

```diff
@@ -178,7 +178,7 @@ function map_glossary_entries_to_translation_originals( $translation, $glossary
 			$glossary_entries_reference[ $term ][] = $id;
 		}
 
-		$terms_search = '\b(';
+		$terms_search = '(';
```

Comment:
I don't think we can remove this as we'll otherwise get partial matches which will be confusing. For an example a glossary entry for "test" is matched in "latest".


<img width="338" alt="Bildschirmfoto 2023-06-08 um 11 11 21" src="https://github.com/GlotPress/GlotPress/assets/617637/f7fee464-67d5-4a44-94a8-506846203a87">

Reply:
Right. Will have to add this case to the tests also, and refactor the regex.

Reply:
@ocean90 this code has suffered major changes in the develop branch. I've fixed your suggestion, but I'll wait for the review of https://github.com/GlotPress/GlotPress/pull/1733 that is essencial for this PR to pass the tests.

Reply:
Resolving as the base code changed in #1733